### PR TITLE
photini: init at 2024.2.1

### PR DESCRIPTION
**Pull request by zebreus**
## Description of changes

[Photini](https://github.com/jim-easterbrook/Photini) is a graphical image metadata (Exif, IPTC, XMP) editor for Linux and MacOS.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by LordGrimmauld** [comment]
Tested, builds and runs and does indeed affect metadata.

File: pkgs/by-name/ph/photini/package.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  pname = "photini";
+  version = "2024.2.1";
+
+  src = fetchFromGitHub {
+    owner = "jim-easterbrook";
+    repo = "Photini";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-CYuMMySCQ2d0aVqfXHcRjXnH5H59bPtueCKqjso7QmU=";
+  };
+  pyproject = true;
```
[the remark this answers is not in the document]
Comment (zebreus) [reply pkgs/by-name/ph/photini/package.nix:13]: Moved it up to pname and version, as the nixpkgs manual also does it that way

**Review by LordGrimmauld — APPROVED** [review]
I see pyproject things were fixed, i don't see more issues myself now.
I can't quite tell why ofborg is mad about "This PR does not cleanly list package outputs after merging", and if someone does know whats going on here do tell.

**Comment by zebreus** [comment]
I think the ofborg error is caused by `requests_toolbelt` and `requests_oauthlib` being aliases for `requests-toolbelt` and `requests-oauthlib`. Replacing them with the real package names should fix that issue.

**Review by LordGrimmauld — APPROVED** [review]
Alright, nothing left that looks weird to me. Nice work.

File: pkgs/by-name/ph/photini/package.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+python3Packages.buildPythonApplication rec {
+  pname = "photini";
+  version = "2024.2.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "jim-easterbrook";
+    repo = "Photini";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-CYuMMySCQ2d0aVqfXHcRjXnH5H59bPtueCKqjso7QmU=";
```
Comment (SuperSandro2000) [inline pkgs/by-name/ph/photini/package.nix:12]:
```suggestion
    hash = "sha256-CYuMMySCQ2d0aVqfXHcRjXnH5H59bPtueCKqjso7QmU=";
```

File: pkgs/by-name/ph/photini/package.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  };
+
+  build-system = with python3Packages;  [
+    setuptools-scm
+    toml
```
Comment (SuperSandro2000) [inline pkgs/by-name/ph/photini/package.nix:17]:
```suggestion
```
Iirc the toml feature is automatically enabled and adding that here and upstream is redundant

File: pkgs/development/python-modules/exiv2/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    owner = "jim-easterbrook";
+    repo = "python-exiv2";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-DX0pg80fOSkWqrIvcye0btZGglnizzM9ZEuVEpnEJKQ=";
+  };
+
+  build-system = [
+    pkg-config
+    setuptools
+    toml
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/exiv2/default.nix:25]:
```suggestion
```

File: pkgs/development/python-modules/exiv2/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  nativeCheckInputs = [ unittestCheckHook ];
+  unittestFlagsArray = [ "-s" "tests" "-v" ];
+
+  meta = with lib; {
+    description = "Low level Python interface to the Exiv2 C++ library";
+    homepage = "https://github.com/jim-easterbrook/python-exiv2";
+    changelog = "https://python-exiv2.readthedocs.io/en/release-${version}/misc/changelog.html";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ zebreus ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/exiv2/default.nix:41]:
```suggestion
```
That's wrong. Python defaults to Unix

File: pkgs/by-name/ph/photini/package.nix [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/jim-easterbrook/Photini";
+    changelog = "https://photini.readthedocs.io/en/release-${version}/misc/changelog.html";
+    description = "An easy to use digital photograph metadata (Exif, IPTC, XMP) editing application";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ zebreus ];
+    mainProgram = "photini";
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ph/photini/package.nix:43]:
```suggestion
```

File: pkgs/development/python-modules/exiv2/default.nix (hunk 5 above)
Comment (zebreus) [reply pkgs/development/python-modules/exiv2/default.nix:41]: Thanks, didn't know that. I will just apply the suggestion and remove platforms altogether.

File: pkgs/by-name/ph/photini/package.nix (hunk 3 above)
Comment (zebreus) [reply pkgs/by-name/ph/photini/package.nix:17]: Yes, seems to work without toml here. Weird, I thought I checked it before and it didn't work without it.

**Pull request merged**